### ISSUE TITLE
Regression - Cypress executor does not respect env cli parameters.

### ISSUE THREAD

**Issue by danr-za** [nrwl/nx#10808]
## Current Behavior
Regression of https://github.com/nrwl/nx/issues/9764.
I upgraded from 14.2.4 where it worked.

## Steps to Reproduce

This issue may not be prioritized if details are not provided to help us reproduce the issue.

### Environment
```sh
   Node : 16.13.0
   OS   : darwin x64
   yarn : 3.2.1
   
   nx : 14.3.6
   @nrwl/angular : Not Found
   @nrwl/cypress : 14.3.6
   @nrwl/detox : Not Found
   @nrwl/devkit : 14.3.6
   @nrwl/eslint-plugin-nx : 14.3.6
   @nrwl/express : Not Found
   @nrwl/jest : 14.3.6
   @nrwl/js : 14.3.6
   @nrwl/linter : 14.3.6
   @nrwl/nest : Not Found
   @nrwl/next : Not Found
   @nrwl/node : 14.3.6
   @nrwl/nx-cloud : Not Found
   @nrwl/nx-plugin : Not Found
   @nrwl/react : 14.3.6
   @nrwl/react-native : Not Found
   @nrwl/schematics : Not Found
   @nrwl/storybook : 14.3.6
   @nrwl/web : 14.3.6
   @nrwl/workspace : 14.3.6
   typescript : 4.7.4
   ---------------------------------------
   Community plugins:
         nx-stylelint: 13.4.0

```

**Comment by barbados-clemens** [comment]
@danr-za I just tried this locally and it works correctly.
the only issue is the args nx shows is showing `[object Object]` vs the actual argument but the args are sent to cypress correctly.

https://user-images.githubusercontent.com/23272162/175391323-016fb307-13e7-42c2-8dab-e628d939667c.mp4

**Comment by danr-za** [comment]
@barbados-clemens so this is what I see:
* Our CI command currently looks like this: 
```sh 
yarn app:e2e:production --browser=chrome --baseUrl=<URL> --env.NX_API_URL=<API URL> 
--record -- --parallel --ci-build-id=<build> --key=<key>
```

When running that, I get:
```sh
 NX   Nx didn't recognize the following args: browser, baseUrl, env, record

   When using '--' all executor args have to be defined after '--'.
```

The reason we append it is because of https://github.com/nrwl/nx/issues/8014

Moving it to the beginning results in:
```sh
yarn app:e2e:production -- --browser=chrome --baseUrl=<URL> --env.NX_API_URL=<API URL>
 --record --parallel --ci-build-id=<build> --key=<key>
```

Here, for some reason, the NX_API_URL is not set and is using the default one from the cypress.env.json file.

I will try running without the additional dashes and update.

**Comment by danr-za** [comment]
Running this:
```sh
nx run management-e2e:e2e:production --browser=chrome --baseUrl=http://10.11.103.227/ --env.NX_API_URL=http://10.1.2.3 --record=true --ci-build-id=1234 --key=12345578
```

Results with:
```sh
You passed the --ci-build-id flag but did not provide either a --group or --parallel flag.
```

So there's an issue anyway.

**Comment by danr-za** [comment]
@barbados-clemens any news about this? thanks

**Comment by barbados-clemens** [comment]
that error doesn't look like it's related to the issue you initially raised. I've yet to be able to reproduce this locally as I demoed in the above comments. 

```
You passed the --ci-build-id flag but did not provide either a --group or --parallel flag.
```
I'm not seeing your command pass either flag.

**Comment by rodrigomata** [comment]
I've run randomly into this issue too within Github Actions, seems that additional flags are randomly being ignored. I've also tried removing parallel without success. This began happening suddenly, so it seems to be a regression on NX

**Comment by danr-za** [comment]
Any news about this?

**Comment by danr-za** [comment]
@barbados-clemens - it's still labeled as waiting for repro, is there something else you want me to provide?

**Comment by barbados-clemens** [comment]
Hey all, so just a status update. this is next on my list of things to look more into after #10857 which should be soon as I found the root cause of my last issue. 

but last time looking at myself and @FrozenPandaz were not able to reproduce this locally. so any extra info you can give about the environment and versions would be helpful. also, double checking with the latest nx version with cypress 10 support would also be helpful. nx v14.5.1

**Comment by danr-za** [comment]
Unfortunately, I already tried with the latest version and Cypress 10, but it seems that it has something to do with the CLI args not passing.
Specifically, I guess `parallel` flag is omitted because Nx uses it for the runners.

This breaks in:
* Github actions
* Jenkins

But as I previously mentioned, locally on my M1 Mac it seems to be working (even though I couldn't fully check that as I don't have the access to the env I need)

**Comment by ureciocais** [comment]
> Hey all, so just a status update. this is next on my list of things to look more into after #10857 which should be soon as I found the root cause of my last issue.
> 
> but last time looking at myself and @FrozenPandaz were not able to reproduce this locally. so any extra info you can give about the environment and versions would be helpful. also, double checking with the latest nx version with cypress 10 support would also be helpful. nx v14.5.1

I'm also not only able to reproduce it on my Mac, but I can on Github actions using NX Cloud and distributed execution. If you need any extra info, let me know (I'm currently passing 3 --env args)

**Comment by meeroslav** [comment]
We managed to reproduce it also for one client. Again, locally (Mac, Windows) works fine, but fails on CI (Azure with hosted Ubuntu).

Not sure how we can test/debug this. Any ideas @barbados-clemens?

**Comment by barbados-clemens** [comment]
So I just set this up in a GitHub action on Ubuntu latest and it worked https://github.com/barbados-clemens/testing-validation-bench/runs/7808421613?check_suite_focus=true

Gonna keep looking into this but might be more going on than just OS

**Comment by barbados-clemens** [comment]
Status update. this looks to be related to DTE?
I enabled DTE on my test repo, and finally got the error @meeroslav ran into 
` >  NX   Property 'env' does not match the schema. '[object' should be a 'object'.`

see cloud run: https://nx.app/runs/8nkrV8kxzNP

@ureciocais mentioned they have DTE on as well. 

@danr-za and @rodrigomata are any of y'all using nx cloud + DTE with this error?

**Comment by danr-za** [comment]
Hmm not us

**Comment by barbados-clemens** [comment]
@danr-za are you getting any errors or just env vars being undefined with the usage of `Cypress.env('my-var)`?

**Comment by danr-za** [comment]
An update on my side:
I have moved `parallel` and `record` to a ci configuration as you suggested (which should have been done from the beginning as they have constant values) and everything seems to work now, also the env variables passed into Cypress.
I assume that they are only logged this way `--env=[object Object]` but actually are passed correctly.

**Comment by barbados-clemens** [comment]
@danr-za that's very interesting. presumably that should have just worked anyways, and also the fact it's influencing the --env params is also interesting. I'll investigate some on my side to see what I find, thanks for more info. 

> I assume that they are only logged this way --env=[object Object] but actually are passed correctly.

Correct, they will be still be passed to the executors as expected, unless you use DTE, where they are currently getting mangled, but presumably [this recent change made will fix that](https://github.com/nrwl/nx/pull/11564). 

I also [have a PR up](https://github.com/nrwl/nx/pull/11569) to document usage patterns with env that adds using `--env.<value>` to the cypress e2e tests so we can prevent any other regressions.

**Comment by acomben-cais** [comment]
The recent change doesn't appear to fix the issue

**Comment by ureciocais** [comment]
@barbados-clemens, I've tried with the latest version, but still arguments aren't passed in when using DTE

**Comment by barbados-clemens** [comment]
@acomben-cais @ureciocais what are the exact values you are passing through the command line?

**Comment by ureciocais** [comment]
@barbados-clemens we are doing something like this:
`yarn nx affected:e2e --base=remotes/origin/main --head=HEAD --parallel=1 --browser=chrome --env.E2E_USERNAME=${{ secrets.E2E_USERNAME }} --env.E2E_PASSWORD=${{ secrets.E2E_PASSWORD }} --env.E2E_AUTHO_CLIENT_SECRET=${{ secrets.E2E_AUTHO_CLIENT_SECRET }}`

**Comment by barbados-clemens** [comment]
@ureciocais can you run `nx report` and post the info here?

**Comment by barbados-clemens** [comment]
Oops, not stale. sorry about that.

**Comment by lukasalphta** [comment]
I am facing the same issue with ``` [object' should be a 'object'.``` when run in CI with DTE. 
In my case I am passing ```--env.grepTags="tag1 tag2"``` to the nx.

[11564](https://github.com/nrwl/nx/pull/11564) did not get rid completely of ```[object Object]``` in terminal output so I created [another PR ](https://github.com/nrwl/nx/pull/12499) which might fix the problem. Would appreciate if it can be reviewed soon.

**Comment by lukasalphta** [comment]
@barbados-clemens  @meeroslav In my [PR](https://github.com/nrwl/nx/pull/12499) I adjusted also the way how the section "With additional flags" will print the arguments as I think they should be printed in unparsed way mirroring the actual input.

So this way:
```
         With additional flags:
           --env.cliArg="i am from the cli args"
```
rather than
```
         With additional flags:
           --env={"cliArg": "i am from the cli args"} 
```
Maybe somebody of you will be willing to have a look at this?
